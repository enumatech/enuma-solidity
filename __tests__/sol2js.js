const path = require('path');
const fs = require('fs');
const sol2js = require('../');

function testFixture(name, done) {
  const fixture = path.join(__dirname, '../__fixtures__/', `${name}.sol`);
  const sol = fs.readFileSync(fixture, 'utf8');

  console.log(sol);

  sol2js(sol, 'ws://localhost:7545', (err, js) => {
    expect(js).toMatchSnapshot();
    done();
  });
}

test('Hello World', done => {
  testFixture('HelloWorld', done);
});
