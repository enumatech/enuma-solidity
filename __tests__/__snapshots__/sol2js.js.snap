// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World 1`] = `
"import Web3 from 'web3';
  import assert from 'assert';
  const { validators } = require('../src/utils');

  const web3 = new Web3(\\"ws://localhost:7545\\");

  
  export class HelloWorld {

      /**
       *
       * @param {string} address
       */
      constructor(address){
          assert.equal(arguments.length, 1);

          this._contract = new web3.eth.Contract([{\\"constant\\":true,\\"inputs\\":[],\\"name\\":\\"Hi\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"string\\"}],\\"payable\\":false,\\"stateMutability\\":\\"pure\\",\\"type\\":\\"function\\"}], address);
      }

      
  /**

* @return {string}*/
  Hi() {
    assert.equal(arguments.length, 0);
    
    return this._contract.methods.Hi();
  }
  
  }"
`;
